src/Mvc/Mvc.Razor/src/Diagnostics/MvcDiagnostics.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.Diagnostics
+{
+    public sealed class BeforeViewPage : EventData
+    {
+        public const string EventName = EventNamespace + 
+            "Razor." +
+            nameof(BeforeViewPage);
+
+        public BeforeViewPage(IRazorPage page, ViewContext viewContext, ActionDescriptor actionDescriptor, HttpContext httpContext)
+        {
+            Page = page;
+            ViewContext = viewContext;
+            ActionDescriptor = actionDescriptor;
+            HttpContext = httpContext;
+        }
+
+        public IRazorPage Page { get; }
+        public ViewContext ViewContext { get; }
+        public ActionDescriptor ActionDescriptor { get; }
+        public HttpContext HttpContext { get; }
+
+        protected override int Count => 4;
+
+        protected override KeyValuePair<string, object> this[int index] => index switch
+        {
+            0 => new KeyValuePair<string, object>(nameof(Page), Page),
+            1 => new KeyValuePair<string, object>(nameof(ViewContext), ViewContext),
+            2 => new KeyValuePair<string, object>(nameof(ActionDescriptor), ActionDescriptor),
+            3 => new KeyValuePair<string, object>(nameof(HttpContext), HttpContext),
+            _ => throw new IndexOutOfRangeException(nameof(index))
+        };
+    }
+
+    public sealed class AfterViewPage : EventData
+    {
+        public const string EventName = EventNamespace +
+            "Razor." +
+            nameof(AfterViewPage);
+
+        public AfterViewPage(IRazorPage page, ViewContext viewContext, ActionDescriptor actionDescriptor, HttpContext httpContext)
+        {
+            Page = page;
+            ViewContext = viewContext;
+            ActionDescriptor = actionDescriptor;
+            HttpContext = httpContext;
+        }
+
+        public IRazorPage Page { get; }
+        public ViewContext ViewContext { get; }
+        public ActionDescriptor ActionDescriptor { get; }
+        public HttpContext HttpContext { get; }
+
+        protected override int Count => 4;
+
+        protected override KeyValuePair<string, object> this[int index] => index switch
+        {
+            0 => new KeyValuePair<string, object>(nameof(Page), Page),
+            1 => new KeyValuePair<string, object>(nameof(ViewContext), ViewContext),
+            2 => new KeyValuePair<string, object>(nameof(ActionDescriptor), ActionDescriptor),
+            3 => new KeyValuePair<string, object>(nameof(HttpContext), HttpContext),
+            _ => throw new IndexOutOfRangeException(nameof(index))
+        };
+    }
+
+    public sealed class BeginInstrumentationContext : EventData
```

Comment:
If possible, I'd prefer not to define these events. I don't think we actually fire them anymore.

Reply:
Removed